realm/realm-library/src/main/java/io/realm/internal/TableView.java

```diff
@@ -68,6 +68,19 @@ protected TableView(Context context, Table parent, long nativePtr, TableQuery qu
         this.query = query;
     }
 
+    /**
+     * Creates a copy of already created Java TableView Object.
+     * The method is not supposed to be called by the user of the db. The method is for internal use only.
+     *
+     * @param tableView An existing TableView
+     */
+    protected TableView(TableView tableView) {
+        this.context = tableView.context;
+        this.parent = tableView.parent;
+        this.nativePtr = nativeGetCopy(tableView.nativePtr);
+        this.query = tableView.query;
```

Comment:
This seems a bit dangerous, as that query reference is now used in two tableviews that might be used independently. 
This should probably be refactored to something like `nativeGetQuery()` or something like that (not sure if it already exists).